docs/root/intro/arch_overview/http/http_proxy.rst

```diff
@@ -40,15 +39,20 @@ For further configuration information see the :ref:`HTTP filter configuration do
 Memory usage details
 --------------------
 
-.. attention::
-
-  HTTP dynamic forward proxy support currently can use unbounded memory and is not ready for
-  production use. Circuit breakers and other limits will be added soon.
-
 Memory usage detail's for Envoy's dynamic forward proxy support are as follows:
 
 * Each resolved host/port pair uses a fixed amount of memory global to the server and shared
   amongst all workers.
 * Address changes are performed inline using read/write locks and require no host reallocations.
 * Hosts removed via TTL are purged once all active connections stop referring to them and all used
   memory is regained.
+* The :ref:`max_hosts
+  <envoy_api_field_config.common.dynamic_forward_proxy.v2alpha.DnsCacheConfig.max_hosts>` field can
+  be used to limit the number of hosts that the DNS cache will store at any given time.
+* The cluster's :ref:`max_pending_requests
+  <envoy_api_field_cluster.CircuitBreakers.Thresholds.max_pending_requests>` circuit breaker can
+  be used to limit the number of requests that are pending waiting for the DNS cache to load
+  a host.
+* Long lived upstream connections can have the underlying logical host expire via TTL while the
+  connection is still open. Upstream requests and connections are still bound by other cluster circuit breakers
```

Comment:
Maybe mention the lack of explicit bounds of hosts here as well?

Reply:
The more I think about this, I really don't think this matters other than in a multi-tenant environment in which the cache is shared between clusters. This is because referring to host data via shared pointer, even after it has been ejected from the cache, is marginal above the existing per-connection memory usage, and we already have circuit breakers on connections and requests. I think above you are saying that you think this is an issue in non-shared environments and I think now I'm not sure that's the case. :)

Reply:
So basically the marginal cost of the host above the HCM and stream memory is basically moot?
That makes sense to me.  May be worth documenting but I think I'm good with this as-is then.

Reply:
yeah, that's what I think. let me take another crack at some comments, etc.